### PR TITLE
chore: remove React 16.4's obsolete React imports

### PR DESCRIPTION
**Pull request by hainenber**
chore: remove unused React imports

### SUMMARY

React imports are no longer necessary starting from React 16.4.0 ([source](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#removing-unused-react-imports)), I ran `npx react-codemod update-react-imports`. Afterwards,  I executed this snippet to confirm everything stays intact
```
npm ci
npm run build
npm run test
npm run dev-server
```

Notable changes:
- Added ESLint configs as [recommended](https://legacy.reactjs.org/blog/2020/09/22/introducing-the-new-jsx-transform.html#eslint) per React doc
- Added Babel and Webpack configs to allow working with new automatic JSX runtime.
- Fix a build warning due to a type import misdirected as normal import.

### TESTING INSTRUCTIONS

All CI should be green

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by hainenber** [comment]
Converting this to `draft` as there are weird behaviors after removal

**Comment by hainenber** [comment]
Phew! I've worked out all the edge cases and CI is green now, re-open for review :D

**Comment by hainenber** [comment]
I've added the lint rule as requested :D

File: superset-frontend/babel.config.js [hunk 1]
```diff
@@ -74,7 +78,6 @@ module.exports = {
             targets: { node: 'current' },
           },
         ],
-        ['@emotion/babel-preset-css-prop'],
```
Comment (rusackas) [inline superset-frontend/babel.config.js:77]: We don't need this any more?

File: superset-frontend/packages/superset-ui-core/src/chart/components/reactify.tsx [hunk 2]
```diff
@@ -17,7 +17,8 @@
  * under the License.
  */
 
-import React from 'react';
+// eslint-disable-next-line no-restricted-syntax -- whole React import is required for `reactify.test.tsx` Jest test passing.
```
Comment (rusackas) [inline superset-frontend/packages/superset-ui-core/src/chart/components/reactify.tsx:20]: Appreciate the comments!

File: superset-frontend/babel.config.js (hunk 1 above)
Comment (hainenber) [reply superset-frontend/babel.config.js:77]: Yes, this is included in `@emotion/babel-plugin` already for new JSX runtime ([source](https://www.npmjs.com/package/@emotion/babel-preset-css-prop))

I already added it [here](https://github.com/apache/superset/pull/28571/files#diff-5cc6b0da6915b09d0c2d20aa46e0196b1daced2e21e726f404d31ec4579f97b5R59) :D

File: superset-frontend/packages/superset-ui-core/src/chart/components/reactify.tsx (hunk 2 above)
Comment (hainenber) [reply superset-frontend/packages/superset-ui-core/src/chart/components/reactify.tsx:20]: I should clarify that I was unable to refactor the `reactify.test.tsx` to decouple it from whole React import :(

I hope someone much better than me can do the needful in the future.

**Review by rusackas — APPROVED** [review]
Trying to review this big of a PR is giving my browser a heart attack, and making my eyes bleed :P 

I've reviewed about 700 of the files in a somewhat random order, and all looks good to me.

GitHub makes me "load diff" for the rest of the files, and it takes foreeeeeeeever. You would think filtering by file type or filtering out "viewed" files would help, but NOPE.

**Review by michael-s-molina — APPROVED** [review]

**Pull request merged**